reporthandling/helpers/v1/listing.go

```diff
@@ -1,145 +1,154 @@
+func (all *AllLists) Passed() int  { return all.passed }
+func (all *AllLists) Skipped() int { return all.skipped }
+func (all *AllLists) Other() int   { return all.other }
+func (all *AllLists) Len() int {
+	return all.failed + all.passed + all.skipped + all.other
 }
-
-type AllListsIterator struct {
-	allLists     *AllLists
-	size         int
-	index        int
-	failedIndex  int
-	passIndex    int
-	skippedIndex int
-	otherIndex   int
+func (all *AllLists) All() map[string]apis.ScanningStatus {
+	return all.itemToStatus
 }
 
-func (all *AllLists) createIterator() Iterator {
-	return &AllListsIterator{
-		size:     len(all.failed) + len(all.passed) + len(all.skipped) + len(all.other),
-		allLists: all,
+// Initialize initialize the AllLists object map with the given size - this is an optimization for the map
+func (all *AllLists) Initialize(size int) {
+	if all.itemToStatus == nil {
+		all.itemToStatus = make(map[string]apis.ScanningStatus, size)
 	}
 }
 
-func (iter *AllListsIterator) Len() int {
-	return iter.size
+// Clear remove all items and reset the counters
+func (all *AllLists) Clear() {
+	if all.itemToStatus != nil {
+		maps.Clear(all.itemToStatus)
+		all.passed = 0
+		all.failed = 0
+		all.skipped = 0
+		all.other = 0
+	}
 }
 
-func (iter *AllListsIterator) HasNext() bool {
-	return iter.index < iter.size
-}
+// Append append single string to matching status list
+func (all *AllLists) Append(status apis.ScanningStatus, str ...string) {
+	if all.itemToStatus == nil {
+		all.itemToStatus = make(map[string]apis.ScanningStatus, len(str))
+	}
 
-func (iter *AllListsIterator) Next() string {
-	var item string
-	if iter.HasNext() {
-		if iter.failedIndex < len(iter.allLists.failed) {
-			item = iter.allLists.failed[iter.failedIndex]
-			iter.failedIndex++
-		} else if iter.passIndex < len(iter.allLists.passed) {
-			item = iter.allLists.passed[iter.passIndex]
-			iter.passIndex++
-		} else if iter.skippedIndex < len(iter.allLists.skipped) {
-			item = iter.allLists.skipped[iter.skippedIndex]
-			iter.skippedIndex++
-		} else if iter.otherIndex < len(iter.allLists.other) {
-			item = iter.allLists.other[iter.otherIndex]
-			iter.otherIndex++
+	for _, s := range str {
+		oldStatus, exist := all.itemToStatus[s]
+		if !exist {
+			all.itemToStatus[s] = status
+			switch status {
+			case apis.StatusPassed:
+				all.passed++
+			case apis.StatusFailed:
+				all.failed++
+			case apis.StatusSkipped:
+				all.skipped++
+			default:
+				all.other++
+			}
+			// element exist with different status
+		} else if oldStatus != status {
+			// check if the new status is more significant
+			if result := apis.Compare(oldStatus, status); result == status {
+				all.itemToStatus[s] = status
+				switch status {
+				case apis.StatusPassed:
+					all.passed++
+				case apis.StatusFailed:
+					all.failed++
+				case apis.StatusSkipped:
+					all.skipped++
+				default:
+					all.other++
+				}
+
+				// update the old status
+				switch oldStatus {
```

Comment:
this logic fits a new function to get the status and int to append ( the count with some int (+/-))

Reply:
Fixed in https://github.com/kubescape/opa-utils/pull/108